src/ORM/FieldType/DBGeneratedColumn.php

```diff
@@ -0,0 +1,288 @@
+<?php
+
+namespace SilverStripe\ORM\FieldType;
+
+use InvalidArgumentException;
+use SilverStripe\Assets\Storage\DBFile;
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Forms\FormField;
+use SilverStripe\Model\ModelData;
+use SilverStripe\ORM\DB;
+use SilverStripe\ORM\Filters\SearchFilter;
+
+/**
+ * A column which gets generated based on some SQL statement.
+ * This can be stored (generated whenever columns in the generation statement are updated), or virtual (generated on the fly when accessed).
+ */
+class DBGeneratedColumn extends DBField
+{
+    public const string GENERATION_STORED = 'STORED';
+
+    public const string GENERATION_VIRTUAL = 'VIRTUAL';
+
+    /**
+     * DBField classes which must not be used as the underlying data representation for a generated column.
+     */
+    private static array $invalid_child_field_types = [
+        DBClassName::class,
+        DBClassNameVarchar::class,
+        DBComposite::class,
+        DBFile::class,
+        DBForeignKey::class,
+        DBGeneratedColumn::class,
+        DBPolymorphicForeignKey::class,
+        DBPrimaryKey::class,
+    ];
+
+    private DBField $childField;
+
+    private string $generationExpression;
```

Comment:
```suggestion
    private string $sqlExpression;
```
And update references

Reply:
Done